### ISSUE TITLE
Don't require YAML file

### ISSUE THREAD

**Issue by fbreitwieser** [ramnathv/htmlwidgets#304]
If the file `htmlwidgets/WIDGET.yaml` does not exist, the following error is thrown:
```
Error in yaml.load(readLines(con), error.label = error.label, ...) : 
  argument "error.label" is missing, with no default
```

However, since dependencies can be specified with `createWidgets`, no yaml file is required.

**Comment by ramnathv** [comment]
@fbreitwieser I would like to understand your use case better. The `dependencies` argument in `createWidgets` was added to support dynamic dependencies. Are you encountering a situation where all dependencies are dynamic and hence you don't want to specify a YAML file?

![image](https://user-images.githubusercontent.com/346288/37792929-68e7ba40-2dca-11e8-8c1e-c719aae0b0b2.png)

**Comment by fbreitwieser** [comment]
Thanks for the quick answer. Personally I just found it easier to define all dependencies using the `dependencies` argument - even the static ones. When using `d3r` for example, I'd define the d3 dependencies with `d3r::d3_dep_v4()`.

**Comment by ramnathv** [comment]
@fbreitwieser That makes sense. It should not be hard modifying the logic so that when a widget yaml file is absent, things work differently.

**Comment by timelyportfolio** [comment]
closed by #306; feel free to reopen for discussion or issues
